### PR TITLE
Add support to arm64 at Dockerfile

### PR DESCRIPTION
**Pull request by mshima**
Related to https://github.com/jhipster/generator-jhipster/issues/15881.

- Switches docker JRE image from now deprecated adoptopenjdk to eclipse-temurin. See https://github.com/AdoptOpenJDK/openjdk-docker#-deprecation-notice-
- ~Use node 16 at arm64 image.~
- Update node to 14.17.6

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by mraible — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -19,14 +20,24 @@ RUN \
...
+       ;; \
+     amd64|x86_64) \
+       ARCH='x64'; \
+       ;; \
+     *) \
+       echo "Unsupported arch: ${ARCH}"; \
+       exit 1; \
+       ;; \
+  esac; \
+  wget https://nodejs.org/dist/v14.17.6/node-v14.17.1-linux-$ARCH.tar.gz -O /tmp/node.tar.gz && \
```
Comment (DanielFran) [inline Dockerfile:37]:
```suggestion
  wget https://nodejs.org/dist/v14.17.6/node-v14.17.6-linux-$ARCH.tar.gz -O /tmp/node.tar.gz && \
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Review by DanielFran — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -19,14 +20,24 @@ RUN \
...
+     *) \
+       echo "Unsupported arch: ${ARCH}"; \
+       exit 1; \
+       ;; \
+  esac; \
+  wget https://nodejs.org/dist/v14.17.6/node-v14.17.6-linux-$ARCH.tar.gz -O /tmp/node.tar.gz && \
   tar -C /usr/local --strip-components 1 -xzf /tmp/node.tar.gz && \
   # upgrade npm
-  npm install -g npm && \
+  npm install -g npm@latest && \
```
Comment (DanielFran) [inline Dockerfile:40]: @mshima Why use latest version of npm?
Better to use lts, no?

Comment (mshima) [reply Dockerfile:40]:
```suggestion
  npm install -g npm@^7 && \
```
`npm install -g npm` doesn’t update npm currently, don’t remember npm@6 behavior.
Just update from npm 6 to 7 is enough.

**Comment by DanielFran** [comment]
@mshima please fix the conflicts

**Pull request merged**